cyberdrop_dl/crawlers/leakedzone.py

```diff
@@ -0,0 +1,123 @@
+from cyberdrop_dl.utils.m3u8 import M3U8Media
+from cyberdrop_dl.utils.utilities import error_handling_wrapper, get_text_between
+
+if TYPE_CHECKING:
+    from bs4 import BeautifulSoup
+
+    from cyberdrop_dl.data_structures.url_objects import ScrapeItem
+
+
+class PostType(IntEnum):
+    IMAGE = 0
+    VIDEO = 1
+
+
+class Selectors:
+    JS_PLAYER = "script:contains('playerInstance.setup')"
+    MODEL_NAME_COLLECTION = "div.actor-name > h1"
+    MODEL_NAME = "h2.actor-title-port"
+
+
+_SELECTORS = Selectors()
+
+PRIMARY_URL = AbsoluteHttpURL("https://leakedzone.com")
+IMAGES_CDN = AbsoluteHttpURL("https://image-cdn.leakedzone.com/storage/")
+
+
+class LeakedZoneCrawler(Crawler):
+    SUPPORTED_PATHS: ClassVar[SupportedPaths] = {
+        "Video": "/<model_id>/video/<video_id>",
+        "Model": "/<model_id>",
+    }
+    DOMAIN: ClassVar[str] = "leakedzone"
+    FOLDER_DOMAIN: ClassVar[str] = "LeakedZone"
+    PRIMARY_URL: ClassVar[AbsoluteHttpURL] = PRIMARY_URL
+
+    def __post_init__(self) -> None:
+        self.request_limiter = AsyncLimiter(3, 10)
+
+    async def fetch(self, scrape_item: ScrapeItem) -> None:
+        if len(scrape_item.url.parts) >= 4 and "video" in scrape_item.url.parts:
+            return await self.video(scrape_item, video_id=scrape_item.url.parts[-1])
+        return await self.collection(scrape_item)
+
+    @error_handling_wrapper
+    async def collection(self, scrape_item: ScrapeItem) -> None:
+        async with self.request_limiter:
+            soup: BeautifulSoup = await self.client.get_soup(self.DOMAIN, scrape_item.url)
+        model_name: str = soup.select_one(_SELECTORS.MODEL_NAME_COLLECTION).get_text(strip=True)
+        title = self.create_title(model_name)
+        scrape_item.setup_as_profile(title)
+
+        req_params = {"params": {"page": 1}}
+        user_agent: str = self.manager.config_manager.global_settings_data.general.user_agent
+        headers = {"user-agent": user_agent, "X-Requested-With": "XMLHttpRequest"}
+        while True:
+            async with self.request_limiter:
+                posts = await self.client.get_json(
+                    self.DOMAIN, scrape_item.url, headers=headers, request_params=req_params
+                )
+            if not posts:
+                break
+            for post in posts:
+                post_type = PostType(post["type"])
+                if post_type == PostType.VIDEO:
+                    await self.handle_gallery_video(scrape_item, post, model_name)
+                elif post_type == PostType.IMAGE:
+                    await self.handle_gallery_image(scrape_item, post)
+            req_params["params"]["page"] += 1
+
+    async def handle_gallery_video(self, scrape_item, post, model_name):
+        video_id: str = str(post["id"])
+        canonical_url: AbsoluteHttpURL = PRIMARY_URL / scrape_item.url.parts[1] / "video" / video_id
+        if await self.check_complete_from_referer(canonical_url):
+            return
```

Comment:
Update the scrape_item url with the canonical url for this check to work.

The referer value in the database is the scrape_item url
```suggestion
        if await self.check_complete_from_referer(canonical_url):
            return
        scrape_item.url = canonical_url
```